### ISSUE TITLE
chore(shared): dívidas de colocação — exceções, convenção de use case, AssetId e tradução de erro

### ISSUE THREAD

**Issue by MarceloSanC** [MarceloSanC/financial-forecasting#69]
## Contexto

Quatro dívidas de colocação levantadas na auditoria. Nenhuma bloqueia Stage
alguma; todas são baratas. Agrupadas numa issue porque separadamente nenhuma
justifica um card, e três delas se resolvem no mesmo commit.

**(a) Exceções de aplicação dentro do domínio.**
`shared/domain/exceptions/base.py` define `ApplicationError` (linha 23) e
`DuplicateKeyError` (linha 31). Colisão de chave é fato de persistência, não de
negócio.

**(b) Convenção de invocação inconsistente.** `__call__` em 5 use cases
(`analytics_store`, `modeling`) e `execute` em 5 (`market_data`,
`feature_engineering`).

**(c) Normalização de identificador de ativo em um único adapter.**
`market_data/adapters/out/parquet/parquet_fundamental_fetcher.py:60-62`:
```python
return asset_id.split(".", maxsplit=1)[0].upper()
```
Ausente nos outros três adapters e ausente do domínio. Se `AAPL`, `aapl` e
`AAPL.SA` são o mesmo ativo, isso é regra de identidade do modelo — o mesmo
defeito da #64, em escala menor. Efeito prático: `IngestFundamentals` grava
`"AAPL"` e `IngestCandles` grava `"aapl"`; o join a jusante devolve candles sem
fundamentos, **em silêncio**.

**(d) Exceções de terceiros atravessando o port.** Os ports declaram
`Raises: ValueError` (`baseline_forecaster.py:70`, `quantile_model_trainer.py:139`,
`tft_trainer.py:226`, `hyperparameter_search.py:125`) e exceções de
`httpx`/LightGBM passam sem tradução. `LightGBMError` descende de `Exception`
(verificado em `lightgbm/basic.py:152`), logo não é capturada por nenhum
`except (ValueError, RuntimeError, TypeError)` a jusante.

## O quê e como

**(a)** Mover `ApplicationError` e `DuplicateKeyError` para
`shared/application/exceptions.py`. `DomainError` fica. `NotFoundError` é
ambíguo — decidir e documentar em uma linha. ~10-15 imports afetados.

**(b)** Padronizar. Recomendação com razão técnica, não estética: se a #68 for
adotada, `__call__` deixa os ports exprimíveis como `Callable[[Command], Result]`,
o que torna fakes de teste em lambdas. Isso desempata a favor de `__call__`. Se
#68 não for adotada, `execute` é mais grepável.

**(c)** VO `AssetId` com `AssetId.parse(raw)` fazendo split + upper + validação;
os quatro adapters passam a usá-lo. **Verificar os dados antes de ligar** — se
algum Parquet já persistido tem `asset_id` não-normalizado, joins mudam
silenciosamente. Com um único ativo no piloto é provável que seja inócuo;
confirmar e seguir. Fazer adapter por adapter, não num commit só.

**(d)** Envolver as chamadas de biblioteca em `try/except` de **tipos
enumerados** — nunca `except Exception`, que troca um problema por outro pior.
Considerar substituir o `ValueError` cru do contrato por exceções nomeadas
(`ForecasterError(DomainError)`, `FetchError`): `ValueError` é fraco porque o
domínio também o levanta, e o chamador não distingue "entrada inválida" de "a lib
explodiu". **Escopar**: começar pelos adapters do caminho confirmatório
(`statsforecast`, `lightgbm`, `pytorch_forecasting`); os de ingestão
(`yfinance`, `alpha_vantage`) podem esperar — falham cedo, ruidosamente, e fora
da cadeia de resultados.

## Fundamentação

- **(a) Evans, LAYERED ARCHITECTURE** (verificado): *"eliminate any dependency on
  infrastructure [...] **or even application logic that is not business logic**."*
  Também **Martin, Dependency Rule** (verificado).
- **(b)** Nenhuma fonte canônica prescreve `__call__` ou `execute`. O mais
  próximo é **Evans, MODULES/INTENTION-REVEALING INTERFACES** — o argumento é de
  *consistência de linguagem*, não de escolha específica.
- **(c) Evans, ENTITIES** (verificado): *"The model must define what it means to
  be the same thing."* **Evans, VALUE OBJECTS** (verificado): *"give it related
  functionality."*
- **(d) Cockburn** (verificado): *"code pertaining to the inside part should not
  leak into the outside part"* — e a regra espelhada é a razão de ser do adapter.
  **Evans, ANTICORRUPTION LAYER** (verificado): *"create an isolating layer to
  provide your system with functionality of the upstream system **in terms of
  your own domain model**"* — este é o lugar onde o ACL de fato se aplica neste
  projeto, e não entre bounded contexts (ver #68).

## Critério de pronto

- [ ] `ApplicationError`/`DuplicateKeyError` fora de `domain/`
- [ ] Convenção única de invocação nos 10 use cases
- [ ] `AssetId` existe e é usado pelos 4 adapters; verificação de dados registrada no PR
- [ ] Adapters do caminho confirmatório traduzem exceção de lib para tipo do contrato
- [ ] Nenhum `except Exception` introduzido
- [ ] `make check` verde

## Não-objetivos

- Tradução de exceção nos adapters de ingestão (fica para depois).
- Redesenhar a hierarquia de exceções além do movimento de camada.

## Momento previsto

**Oportunístico** — quando alguém já estiver no arquivo. (a) e (b) podem ir num
único commit de ~20 minutos. (c) exige verificação de dados antes. (d) é
incremental por adapter.

Exceção: se a #68 for executada, fazer (b) **junto** com ela, porque a decisão
`__call__` vs `execute` depende do desenho dos Protocols.

## Referências

- Issue #64 — mesmo defeito de identidade, em escala maior
- Issue #68 — Shared Kernel (a decisão de (b) depende dela)
- DDD Reference (Evans, 2015): https://www.domainlanguage.com/wp-content/uploads/2016/05/DDD_Reference_2015-03.pdf
- https://alistair.cockburn.us/hexagonal-architecture
